### ISSUE TITLE
Failed to run : The argument 'file' cannot be empty. Received ''

### ISSUE THREAD

**Issue by jost-s** [nix-community/vscode-nix-ide#288]
That's the error message I get in VSCode when I hit cmd+shift+f as usual. It used to work perfectly fine, but it doesn't any longer. I don't know what has changed.

Formatting for other file types with this shortcut works fine.

Does anyone have a clue how I can fix this? Or how to figure out what the problem is?

**Comment by hall** [comment]
What are your extension settings? Specifically `nix.formatterPath` but also `nix.serverPath` if `nix.enableLanguageServer` is `true`.

**Comment by jost-s** [comment]
Mad, that wasn't set at all. I don't remember to ever have set that.

After switching the computer that must have gone lost. Thanks so much, it works now after setting `nix.formatterPath` to `nixpkgs-fmt`.

**Comment by hall** [comment]
You shouldn't have to set `nix.formatterPath` as it [defaults to `nixpkgs-fmt`](https://github.com/nix-community/vscode-nix-ide/blob/a96c25fe9f729007d57e945d060f327c19ef7341/src/configuration.ts#L21).

I'm able to reproduce your error with `nix.formatterPath: ""`; which kinda makes sense, given the space before the colon in `Failed to run :` (as we'd expect some executable name there).

![image](https://user-images.githubusercontent.com/32495955/191842477-230649b9-5e41-49e2-9e6b-a843fbdceced.png)

The empty `file` argument it's talking about is the first argument to [`execFile`](https://nodejs.org/api/child_process.html#child_processexecfilefile-args-options-callback). In our case, that's [derived from `nix.formatterPath`](https://github.com/nix-community/vscode-nix-ide/blob/a96c25fe9f729007d57e945d060f327c19ef7341/src/formatter.ts#L38).

So if you _have_ to set `nix.formatterPath` then I think this is a valid bug but I'll need more info as I'm not able to reproduce without explicitly setting it to an empty string.

**Comment by jost-s** [comment]
What I thought could be the issue is that after migrating to another computer, I didn't have `nixpkgs-fmt` installed. I created that scenario now but the error is a different one:
<img width="453" alt="image" src="https://user-images.githubusercontent.com/28270981/191845447-66ba7416-6ccb-4197-ae87-3b75640e0e37.png">

What might have happened is that I saw this ENOENT error message and then started fiddling with the extension and ended up in the settings file with `nix.formatterPath: ""` awaiting an input but I didn't input anything. Then the error changed to the empty file argument without me noticing the empty formatter path.

If this happened again, I'd reopen this.
